### PR TITLE
libxc: switch to cmake

### PR DESCRIPTION
**Pull request by LecrisUT**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This is needed in order to provide the `LibxcConfig.cmake` files.

Sorry, it has been a while since I've contributed here so I don't remember the guidelines for these, and I don't have a brew environment to do local testings, but I'll try to get others with the environment to do it for me.

### CODE REVIEW

File: Formula/libxc.rb [hunk 1]
```diff
@@ -15,18 +16,17 @@ class Libxc < Formula
...
-    system "./configure", "--prefix=#{prefix}",
-                          "--enable-shared",
-                          "FCCPP=gfortran -E -x c",
-                          "CC=#{ENV.cc}"
-    system "make", "install"
+    system "cmake",
+           "-B", "build",
+           "-DENABLE_FORTRAN=ON",
+           "-DBUILD_SHARED_LIBS=ON",
+           *std_cmake_args
```
Comment (carlocab) [inline Formula/libxc.rb:27]:
```suggestion
    system "cmake", "-S", ".", "-B", "build",
                    "-DENABLE_FORTRAN=ON",
                    "-DBUILD_SHARED_LIBS=ON",
                    *std_cmake_args
```

**Review by carlocab — COMMENTED** [review]
Thanks, one suggestion.

When applying my suggestion, please squash it into your first commit and use the commit message
```
libxc: switch to cmake
```
(or similar). We put the names of formulae first in our commit messages.

**Review by carlocab — COMMENTED** [review]
Thanks, one more suggestion (though this was part of my suggestion earlier too). Could you also squash your commits together please?

**Comment by LecrisUT** [comment]
I thought the squashing was in order change the other commit message. I would have preferred to be squashed in the merge, that way there is still a navigable history in the PRs, but I guess this is not as relevant in formula maintenance

**Review by carlocab — APPROVED** [review]
Thanks!

**Pull request merged**